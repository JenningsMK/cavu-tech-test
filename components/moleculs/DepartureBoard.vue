<template>
  <div class="departure-board">
    <header class="departure-board__header">
      <border-header />
    </header>

    <table-header />

    <ul ref="flights" class="departure-board__flights" aria-live="polite">
      <template v-if="Array.isArray(flightData) && flightData.length">
        <template v-for="flight in flightData">
          <flight-entry :key="flight.flightNumber" :entrys="flight" />
        </template>
      </template>

      <template v-else>
        <p class="loading-message">
          Getting flight data
        </p>
      </template>
    </ul>
  </div>
</template>

<script>
import BorderHeader from '~/components/atoms/headers/BorderHeader';
import TableHeader from '~/components/atoms/headers/TableHeader';
import FlightEntry from '~/components/atoms/listItems/FlightEntry';

export default {
  name: 'LargeDepartureBoard',

  components: {
    BorderHeader,
    TableHeader,
    FlightEntry
  },

  props: {
    flightData: {
      type: Array || String,
      required: true
    }
  }
};
</script>

<style>
  .departure-board {
    background: linear-gradient(90deg, #464646 0%, #000 100%);
    font-size: 1.5rem;
    margin: auto auto 2rem;
    width: 100%;
  }

  .departure-board__header {
    background-image:
      radial-gradient(
        81.65% 10898.53% at 18.35% 50%,
        var(--yellow-v0) 0%,
        var(--yellow-v1) 100%
      );
    padding: var(--padding);
    margin-bottom: 1rem;
  }

  .loading-message {
    color: white;
    line-height: 2;
    text-align: center;
  }

  .departure-board__flights {
    margin: 0 auto;
    padding: 0 1rem 2rem;
  }
</style>
