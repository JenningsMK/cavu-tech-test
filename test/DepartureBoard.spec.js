import {mount} from "@vue/test-utils";
import DepartureBoard from "@/components/moleculs/DepartureBoard";

global.ResizeObserver = jest.fn(() => {
  return {
    observe: () => {
    }
  }
});

describe('The outputted HTML', () => {
  test('We get a total of 4 list elements with the correct out put', () => {
    const comp = mount(DepartureBoard, {
      propsData: {
        flightData: [
          {"scheduledDepartureDateTime":"2022-08-22T10:40:00","scheduledArrivalDateTime":"2022-08-22T01:00:00","estimatedDepartureDateTime":"2022-08-22T11:30:00","actualDepartureDateTime":null,"arrivalAirport":{"name":"Istanbul Airport","cityName":"Istanbul","countryName":"Turkey","code":"IST"},"departureAirport":{"name":"Manchester Airport","cityName":"Manchester","countryName":"United Kingdom","code":"MAN"},"flightNumber":"TK1994","airline":{"name":"Turkish Airlines","code":"TK"},"departureGate":{"name":"Gate 31","number":"31","action":"Final Call"},"arrivalTerminal":null,"departureTerminal":"1","status":"Final Call - Gate 31","scheduledDepartureDateTimeHuman":"10:40","statusClasses":{"closing":true,"boarding":false,"toGate":false,"open":false,"departed":false}},
          {"flightDirection":"departure","scheduledDepartureDateTime":"2022-08-22T10:55:00","scheduledArrivalDateTime":"2022-08-22T01:00:00","estimatedDepartureDateTime":"2022-08-22T11:20:00","actualDepartureDateTime":null,"arrivalAirport":{"name":"Lisbon Portela Airport","cityName":"Lisbon","countryName":"Portugal","code":"LIS"},"departureAirport":{"name":"Manchester Airport","cityName":"Manchester","countryName":"United Kingdom","code":"MAN"},"flightNumber":"TP1317","airline":{"name":"TAP Air Portugal","code":"TP"},"departureGate":{"name":"Gate 22","number":"22","action":"Final Call"},"arrivalTerminal":null,"departureTerminal":"1","status":"Final Call - Gate 22","scheduledDepartureDateTimeHuman":"10:55","statusClasses":{"closing":true,"boarding":false,"toGate":false,"open":false,"departed":false}},
          {"flightDirection":"departure","scheduledDepartureDateTime":"2022-08-22T11:00:00","scheduledArrivalDateTime":"2022-08-22T01:00:00","estimatedDepartureDateTime":"2022-08-22T11:25:00","actualDepartureDateTime":null,"arrivalAirport":{"name":"Orlando International Airport","cityName":"Orlando","countryName":"United States","code":"MCO"},"departureAirport":{"name":"Manchester Airport","cityName":"Manchester","countryName":"United Kingdom","code":"MAN"},"flightNumber":"EI35","airline":{"name":"Aer Lingus","code":"EI"},"departureGate":{"name":"Gate 208","number":"208","action":"Final Call"},"arrivalTerminal":null,"departureTerminal":"2","status":"Final Call - Gate 208","scheduledDepartureDateTimeHuman":"11:00","statusClasses":{"closing":true,"boarding":false,"toGate":false,"open":false,"departed":false}},
          {"flightDirection":"departure","scheduledDepartureDateTime":"2022-08-22T11:00:00","scheduledArrivalDateTime":"2022-08-22T01:00:00","estimatedDepartureDateTime":"2022-08-22T12:40:00","actualDepartureDateTime":null,"arrivalAirport":{"name":"Cancun International Airport","cityName":"Cancun","countryName":"Mexico","code":"CUN"},"departureAirport":{"name":"Manchester Airport","cityName":"Manchester","countryName":"United Kingdom","code":"MAN"},"flightNumber":"TOM192","airline":{"name":"TUI Airways","code":"TOM"},"departureGate":{"name":"Gate 204","number":"204","action":"Gate Open"},"arrivalTerminal":null,"departureTerminal":"2","status":"Go to Gate 204","scheduledDepartureDateTimeHuman":"11:00","statusClasses":{"closing":false,"boarding":false,"toGate":true,"open":false,"departed":false}},
          {"flightDirection":"departure","scheduledDepartureDateTime":"2022-08-22T11:05:00","scheduledArrivalDateTime":"2022-08-22T01:00:00","estimatedDepartureDateTime":"2022-08-22T11:50:00","actualDepartureDateTime":null,"arrivalAirport":{"name":"Brussels Airport","cityName":"Brussels","countryName":"Belgium","code":"BRU"},"departureAirport":{"name":"Manchester Airport","cityName":"Manchester","countryName":"United Kingdom","code":"MAN"},"flightNumber":"SN2174","airline":{"name":"Brussels Airlines","code":"SN"},"departureGate":{"name":"Gate A1","number":"A1","action":"Final Call"},"arrivalTerminal":null,"departureTerminal":"2","status":"Final Call - Gate A1","scheduledDepartureDateTimeHuman":"11:05","statusClasses":{"closing":true,"boarding":false,"toGate":false,"open":false,"departed":false}}
        ]
      }
    });

    const listItems = comp.findAll('.departure-board__flight');

    expect(listItems).toHaveLength(5);
    expect(comp.html()).toMatchSnapshot();
  })
});
