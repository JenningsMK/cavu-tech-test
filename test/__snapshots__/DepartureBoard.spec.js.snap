// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The outputted HTML We get a total of 4 list elements with the correct out put 1`] = `
"<div class=\\"departure-board\\">
  <header class=\\"departure-board__header\\">
    <h1 class=\\"page-title\\">
      Departures
    </h1>
  </header>
  <section class=\\"departure-board__headings\\">
    <div class=\\"grid-alignment\\"><span id=\\"departure\\">Departure time</span> <span id=\\"city-name\\">City Name</span> <span id=\\"code\\">Code</span> <span id=\\"airline\\">Airline</span> <span id=\\"gate\\">Gate</span> <span id=\\"status\\" class=\\"visually-hidden\\">Status</span></div>
  </section>
  <ul aria-live=\\"polite\\" class=\\"departure-board__flights\\">
    <li class=\\"departure-board__flight grid-alignment\\"><time datetime=\\"2022-08-22T10:40:00\\" aria-labelledby=\\"departure\\">
        10:40
      </time> <span aria-labelledby=\\"city-name\\" class=\\"flight__city-name\\">
    Istanbul Airport
  </span> <span aria-labelledby=\\"code\\" class=\\"flight__code\\">
    TK1994
  </span> <span aria-labelledby=\\"airline\\">
    Turkish Airlines
  </span> <span aria-labelledby=\\"gate\\" class=\\"flight__gate\\">
    31
  </span> <span aria-labelledby=\\"status\\" class=\\"flight flight--closing\\">
    Final Call
  </span></li>
    <li class=\\"departure-board__flight grid-alignment\\"><time datetime=\\"2022-08-22T10:55:00\\" aria-labelledby=\\"departure\\">
        10:55
      </time> <span aria-labelledby=\\"city-name\\" class=\\"flight__city-name\\">
    Lisbon Portela Airport
  </span> <span aria-labelledby=\\"code\\" class=\\"flight__code\\">
    TP1317
  </span> <span aria-labelledby=\\"airline\\">
    TAP Air Portugal
  </span> <span aria-labelledby=\\"gate\\" class=\\"flight__gate\\">
    22
  </span> <span aria-labelledby=\\"status\\" class=\\"flight flight--closing\\">
    Final Call
  </span></li>
    <li class=\\"departure-board__flight grid-alignment\\"><time datetime=\\"2022-08-22T11:00:00\\" aria-labelledby=\\"departure\\">
        11:00
      </time> <span aria-labelledby=\\"city-name\\" class=\\"flight__city-name\\">
    Orlando International Airport
  </span> <span aria-labelledby=\\"code\\" class=\\"flight__code\\">
    EI35
  </span> <span aria-labelledby=\\"airline\\">
    Aer Lingus
  </span> <span aria-labelledby=\\"gate\\" class=\\"flight__gate\\">
    208
  </span> <span aria-labelledby=\\"status\\" class=\\"flight flight--closing\\">
    Final Call
  </span></li>
    <li class=\\"departure-board__flight grid-alignment\\"><time datetime=\\"2022-08-22T11:00:00\\" aria-labelledby=\\"departure\\">
        11:00
      </time> <span aria-labelledby=\\"city-name\\" class=\\"flight__city-name\\">
    Cancun International Airport
  </span> <span aria-labelledby=\\"code\\" class=\\"flight__code\\">
    TOM192
  </span> <span aria-labelledby=\\"airline\\">
    TUI Airways
  </span> <span aria-labelledby=\\"gate\\" class=\\"flight__gate\\">
    204
  </span> <span aria-labelledby=\\"status\\" class=\\"flight flight--to-gate\\">
    Gate Open
  </span></li>
    <li class=\\"departure-board__flight grid-alignment\\"><time datetime=\\"2022-08-22T11:05:00\\" aria-labelledby=\\"departure\\">
        11:05
      </time> <span aria-labelledby=\\"city-name\\" class=\\"flight__city-name\\">
    Brussels Airport
  </span> <span aria-labelledby=\\"code\\" class=\\"flight__code\\">
    SN2174
  </span> <span aria-labelledby=\\"airline\\">
    Brussels Airlines
  </span> <span aria-labelledby=\\"gate\\" class=\\"flight__gate\\">
    A1
  </span> <span aria-labelledby=\\"status\\" class=\\"flight flight--closing\\">
    Final Call
  </span></li>
  </ul>
</div>"
`;
